practical-unit-testing-with-rtl/Exercise2/solution/src/hooks/useTicTacToe.ts

```diff
@@ -0,0 +1,64 @@
+import { TicTacToeBoard, BoardSquareValue } from "@utilities/ticTacToe";
+
+import { useCallback, useState } from "react";
+
+import { isGameOver, isWinner } from "../utilities/ticTacToe";
+import { useScore } from "../components/Score";
+
+const createNewTicTacToeBoard = () => Array(9).fill(null);
+
+const fetchWinCheck = async (url: string, data: any) => {
+  const response = await fetch(url, data);
+  const finalData = await response.json();
+  return finalData;
+};
+
+export const useTicTacToe = () => {
+  const { addWinO, addWinX } = useScore();
+  const [board, _setBoard] = useState<TicTacToeBoard>(createNewTicTacToeBoard);
+  const [isX, setIsX] = useState(true);
+  const [winner, setWinner] = useState<BoardSquareValue>(null);
+
+  const [playerNames, setPlayerNames] = useState({});
+
+  const currentTurn: BoardSquareValue = isX ? "X" : "O";
+  const setBoard = (board: any) => {
+    _setBoard(board);
+  };
+  const handleSquareSelect = useCallback(
+    async (selectedSquare: number) => {
+      if (board[selectedSquare]) {
+        return;
+      }
+      const newBoard = [...board];
+      newBoard[selectedSquare] = currentTurn;
+      setBoard(newBoard);
+
+      const winner = isWinner(currentTurn, newBoard);
+      const winCheck = { gameOver: winner, winner: currentTurn };
+
+      if (winCheck.gameOver) {
+        setWinner(winCheck.winner);
+        winCheck.winner === "X" ? addWinX() : addWinO();
```

Comment:
This bit of code seems odd because it’s setting `winCheck` here but not being used elsewhere… I’m guessing this will be changed?